spec/lib/miq_automation_engine/service_methods/miq_ae_service_generic_object_spec.rb

```diff
@@ -0,0 +1,59 @@
+module MiqAeServiceGenericObjectSpec
+  describe MiqAeMethodService::MiqAeServiceGenericObject do
+    let(:generic_object)  { FactoryGirl.create(:generic_object) }
+    let(:service_go)      { MiqAeMethodService::MiqAeServiceGenericObject.find(generic_object.id) }
+    let(:service)         { FactoryGirl.create(:service) }
+    let(:service_service) { MiqAeMethodService::MiqAeServiceService.find(service.id) }
+
+    context "#add_to_service" do
+      it "adds a generic object to service_resources of a valid service" do
+        service_go.add_to_service(service_service)
+        expect(service.service_resources[0].resource_id).to eq(generic_object.id)
+        expect(service.service_resources[0].resource_type).to eq(generic_object.class.name)
+      end
+
+      it "adds a generic object to multiple services" do
+        service_go.add_to_service(service_service)
+
+        service2 = MiqAeMethodService::MiqAeServiceService.find(FactoryGirl.create(:service).id)
+        service_go.add_to_service(service2)
+
+        expect(service.service_resources[0].resource_id).to eq(generic_object.id)
+        expect(service.service_resources[0].resource_type).to eq(generic_object.class.name)
+
+        expect(service2.service_resources[0].resource_id).to eq(generic_object.id)
+        expect(service2.service_resources[0].resource_type).to eq(generic_object.class.name)
+      end
+
+      it "raises an error when adding a generic object to an invalid service" do
+        expect { service_go.add_to_service('wrong type') }
+          .to raise_error(ArgumentError, /service must be a MiqAeServiceService/)
+      end
+    end
+
+    context "#remove_from_service" do
+      it "removes a generic object from a connected service" do
+        service_go.add_to_service(service_service)
+        expect(service.service_resources.count).to eq(1)
+
+        service_go.remove_from_service(service_service)
+        expect(service.service_resources.count).to eq(0)
+      end
+
+      it "removes a generic object from multiple connected service" do
+        service_go.add_to_service(service_service)
+
+        service2 = MiqAeMethodService::MiqAeServiceService.find(FactoryGirl.create(:service).id)
+        service_go.add_to_service(service2)
+
+        expect(service.service_resources.count).to  eq(0)
```

Comment:
@lfu You never call `remove_from_service` in this test which is leading to test failures.  You should setup and verify the counts then remove the generic object from one service and check the counts for both services again.